Src/IronPython/Runtime/Operations/StringOps.cs

```diff
@@ -1732,13 +1721,11 @@ public static class StringOps {
         private static DecoderFallback ReplacementFallback = new DecoderReplacementFallback("\ufffd");
 #endif
 
-        internal static string DoDecode(CodeContext context, IList<byte> s, string errors, string encoding, Encoding e)
-            => DoDecode(context, s, errors, encoding, e, s.Count, out _);
-
-        internal static string DoDecode(CodeContext context, IList<byte> s, string errors, string encoding, Encoding e, int numBytes, out int numConsumed) {
-            byte[] bytes = s as byte[] ?? ((s is Bytes b) ? b.UnsafeByteArray : s.ToArray());
-            int start = GetStartingOffset(e, bytes);
-            int length = numBytes - start;
+        internal static string DoDecode(CodeContext context, ReadOnlyMemory<byte> data, string errors, string encoding, Encoding e) {
+            int start = GetStartingOffset(e, data.Span);
```

Comment:
Since `start` is only used for slicing maybe it could just be a helper (static local) method `RemovePreamble` or something?
```C#
data = RemovePreamble(e, data);
```